examples/object_detection_2d/object_detection_2d/extended/seed_evaluation.py

```diff
@@ -0,0 +1,332 @@
+        iou_threshold=configuration.iou_threshold,
+    )
+
+    return datapoint_metrics(bbox_matches, thresholds)
+
+
+def compute_f1_optimal_thresholds(
+    configuration: ThresholdConfiguration,
+    data: List[Dict],
+) -> Dict:
+    all_bbox_matches = [
+        match_inferences_multiclass(
+            x["gt_bboxes"],
+            filter_inferences(inferences=x["inf_bboxes"], confidence_score=configuration.min_confidence_score),
+            ignored_ground_truths=x["gt_ignored_bboxes"],
+            mode="pascal",
+            iou_threshold=configuration.iou_threshold,
+        )
+        for x in data
+        if not x["inf_ignored"]
+    ]
+
+    optimal_thresholds = compute_optimal_f1_threshold_multiclass(all_bbox_matches)
+    return defaultdict(
+        lambda: configuration.min_confidence_score,
+        optimal_thresholds,
+    )
+
+
+def eval_func(datapoints: pd.DataFrame, inferences: pd.DataFrame, eval_config: ThresholdConfiguration) -> pd.DataFrame:
+    data = pd.DataFrame()
+    data["gt_bboxes"] = datapoints["bboxes"].apply(
+        lambda arr: [LabeledBoundingBox(label=x.label, top_left=x.top_left, bottom_right=x.bottom_right) for x in arr],
+    )
+    data["gt_ignored_bboxes"] = datapoints["ignored_bboxes"]
+    data["inf_bboxes"] = inferences["bboxes"].apply(
+        lambda arr: [
+            ScoredLabeledBoundingBox(score=x.score, label=x.label, top_left=x.top_left, bottom_right=x.bottom_right)
+            for x in arr
+        ],
+    )
+    data["inf_ignored"] = inferences["ignored"]
+    thresholds = get_confidence_thresholds(eval_config, data.to_dict("records"))
+    metrics = data.apply(
+        lambda x: compute_image_metrics(
+            x["gt_bboxes"],
+            x["gt_ignored_bboxes"],
+            x["inf_bboxes"],
+            x["inf_ignored"],
+            eval_config,
+            thresholds,
+        ),
+        axis=1,
+        result_type="expand",
+    ).replace(
+        np.nan,
+        None,
+    )
+    return metrics
+
+
+def seed_evaluation(
+    model_full_name: str,
+    dataset_name: str,
+    groups_df: pd.DataFrame,
+) -> None:
+    # runs the evaluation
+    # deliberately split into 2 invocations for demo purpose, can be combined into one
+    # upload inferences
+    test(dataset_name, model_full_name, infer=get_stored_inferences(groups_df))
+    loaded_datapoints, loaded_inferences = fetch_inferences(dataset_name, model_full_name)
+    print(loaded_inferences.shape)
+
+    # compute and upload datapoint metrics
+    test(
+        dataset_name,
+        model_full_name,
+        eval=eval_func,
+        eval_configs=[
+            ThresholdConfiguration(
+                threshold_strategy=0.3,
+                iou_threshold=0.3,
+                min_confidence_score=0.2,
+            ),
+            ThresholdConfiguration(
+                threshold_strategy="F1-Optimal",
+                iou_threshold=0.5,
+                min_confidence_score=0.0,
+            ),
+        ],
```

Comment:
we might need to update this into
```py
eval_configs=[
            ThresholdConfiguration(
                threshold_strategy=0.3,
                iou_threshold=0.3,
                min_confidence_score=0.2,
            )._to_dict(),
            ThresholdConfiguration(
                threshold_strategy="F1-Optimal",
                iou_threshold=0.5,
                min_confidence_score=0.0,
            )._to_dict(),
        ]
```

or 
```py
eval_configs=[
            dict(
                threshold_strategy=0.3,
                iou_threshold=0.3,
                min_confidence_score=0.2,
            ),
            dict(
                threshold_strategy="F1-Optimal",
                iou_threshold=0.5,
                min_confidence_score=0.0,
            ),
        ]
```

due to [freeform eval config commit](https://github.com/kolenaIO/kolena/pull/252/commits/a1cf901f3d2f9fc42098acca550c61179a04b3c2)